applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -35,32 +35,49 @@ enum state {
 
 static enum state module_state = STATE_DISABLED;
 
+#define LIST_FOR_DVFS_STATE(MACRO, ...)						\
+	MACRO(INITIALIZING, __VA_ARGS__)					\
+	MACRO(LLPM_CONNECTED, __VA_ARGS__)					\
+	MACRO(USB_CONNECTED, __VA_ARGS__)					\
+	MACRO(CONFIG_CHANNEL, __VA_ARGS__)					\
+	MACRO(SMP_TRANSFER, __VA_ARGS__)					\
+	MACRO(COUNT, __VA_ARGS__)
+
+#define DVFS_STATE(name, ...) DVFS_STATE_ ## name,
+
+#define GET_ENUM_STRING(name, ...)						\
+	IF_ENABLED(CONFIG_DESKTOP_DVFS_STATE_ ## name ## _ENABLE,		\
+		(case DVFS_STATE_ ## name: return STRINGIFY(_CONCAT(DVFS_STATE_, name));))
+
+#define _DEFINE_FREQ_MASK_BIT(name, freq_suffix)				\
+	IF_ENABLED(CONFIG_DESKTOP_DVFS_STATE_ ## name ## _ ## freq_suffix,	\
+	(BIT(DVFS_STATE_ ## name) |))
+
+#define INITIALIZE_DVFS_FREQ_MASK(FREQ_SUFFIX)					\
+	LIST_FOR_DVFS_STATE(_DEFINE_FREQ_MASK_BIT, FREQ_SUFFIX) 0
+
+#define INITIALIZE_DVFS_STATE_TIMEOUT(name, ...)				\
+	IF_ENABLED(CONFIG_DESKTOP_DVFS_STATE_ ## name ## _ENABLE,		\
```

Comment:
Since timeout_ms Kconfig configuration option is defined to `0` when timeout is not supported for a given state, we could define initialization members unconditionally here

Reply:
I would leave it as it is

Reply:
bump

Reply:
let's simplify this part

Reply:
We usually do not initialize variables to 0? Why should we do exception here?

Reply:
We also do not typically define configurations that we don't want to use

Reply:
> We usually do not initialize variables to 0? Why should we do exception here? 

to make the code easier to understand

Reply:
We also do not usually add to the bitwise or operation zero `(...) | 0` as this is neutral operation and does not achieve anything

Reply:
> We also do not usually add to the `bitwise OR` operation zero `(...) | 0` as this is a neutral operation and does not achieve anything

don't understand you here. We all know that we need this 0 in case there is no active state. Also it is not argument about topic we have here. This one also:
> We also do not typically define configurations that we don't want to use


In this situation we have case where we can leave it as it is and do not add 0 initialized variables after macro does its job or we can remove macro conditional statement and have it simpler but with initializing variables to "0". I prefer first option you may prefer second. IMO it is sth which isn't worth spending much time on so I can change if you insist. However please use arguments about discussed topic.

We also already agreed that we want to have prompt less config options. Is there right now any sense in going again through this topic?

Reply:
Which of my arguments is not related to this topic?

Two reviewers support this simplification and somehow you are reluctant to introduce this simple (5min) change.

Reply:
> Which of my arguments is not related to this topic?
Those which I mentioned above.

> Two reviewers support this simplification and somehow you are reluctant to introduce this simple (5min) change.
I said that I can change it. Just you said that you want this change without any argument. I provided argument and am waiting for your reply.

Reply:
Also this change doesn't simplify the code because we need to deal with 	MACRO(COUNT, __VA_ARGS__) - without checking we create variable for:

[DVFS_STATE_ COUNT] = {						
		.timeout_ms = CONFIG_DESKTOP_DVFS_STATE_ COUNT_TIMEOUT_MS,
	},

Which of course doesn't exist

Reply:
No, let's leave it. It won't work also for other reasons